### PR TITLE
doc(AM-ContributingGuide): Added contributing guide and related changes

### PR DESCRIPTION
**Pull request by gourishkb**
Makefile has a new chart-undeploy target to delete the deployed worker charts using helm
Contributing guide, Developers' guide and code of conduct added

Signed-off-by: gourishkb <104021126+gourishkb@users.noreply.github.com>

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,55 @@
+# Contributing
+
+Guidelines on how to contribute to KubeSlice
+
+There are a number of ways you can contribute to the KubeSlice project. We would like to accept your patches and contributions to the KubeSlice code base. This document provides guidelines on how to contribute to the KubeSlice code base. we require all participants to adhere to the [Code of Conduct](https://github.com/kubeslice/kubeslice-controller/blob/doc/contributing-guide/code_of_conduct.md).
+
```
Comment (kmjayadeep) [inline CONTRIBUTING.md:6]: this link is broken

Comment (gourishkb) [reply CONTRIBUTING.md:6]: Fixed

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+There are a number of ways you can contribute to the KubeSlice project. We would like to accept your patches and contributions to the KubeSlice code base. This document provides guidelines on how to contribute to the KubeSlice code base. we require all participants to adhere to the [Code of Conduct](https://github.com/kubeslice/worker-operator/blob/doc/contributing-guide/code_of_conduct.md).
+
+
+Here are some ways to contribute:
+
+* Contribute to the kubeslice codebase
+* Report and triage bugs.
+* Answer questions on KubeSlice discussions forums
+* Write technical documentation and blog posts, for users and contributors.
+* Filing issues
```
Comment (veenamj-avesha) [inline CONTRIBUTING.md:14]: For parallelism, change this to `File Issues.`

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+* Provide feedback
+* Submit PRs for small bug fixes
```
Comment (veenamj-avesha) [inline CONTRIBUTING.md:16]: End this sentence with a period/full stop.

Don't they require to submit PRs for any features/enhancements? cc @gourishkb

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+`Signed-off-by: John Doe <jdoe@example.org>`
+
+This can be done by adding `--signoff` to your git command line
+
+## Code of Conduct
+Please make sure to read and observe our [Code of Conduct](https://github.com/kubeslice/worker-operator/blob/doc/contributing-guide/code_of_conduct.md).
+
+## Contributor Workflow
+
+Pull Requests are welcome. In general, we follow the "fork-and-pull" Git workflow.
```
Comment (veenamj-avesha) [inline CONTRIBUTING.md:33]: `In general, we follow the **fork-and-pull** Git workflow that has the following steps: ` 

This sentence follows a list; so change the sentence that starts with `In general...` as above and end with a colon.

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+
+1. **Fork** the repo on GitHub
```
Comment (veenamj-avesha) [inline CONTRIBUTING.md:35]: End all the list items with a period/full stop (1 to 5)

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+2. **Clone** the project to your own machine
+3. **Commit** changes to your own branch
+4. **Push** your work back up to your fork
+5. Submit a **Pull request** so that we can review your changes
+
+NOTE: Be sure to merge the latest from "upstream" before making a pull request!
```
Comment (veenamj-avesha) [inline CONTRIBUTING.md:41]: change "upstream" to **upstream**

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+
+## Filing Issues
+If you encounter a bug please file an issue in the Github repository. Please provide as much information as possible to help fix the issue. Please file the issue under the appropriate repo.
```
Comment (veenamj-avesha) [inline CONTRIBUTING.md:44]: add a comma after `If you encounter a bug`

File: CONTRIBUTING.md [hunk 8]
```diff
@@ -0,0 +1,55 @@
...
+
+
+## Coding Guidelines
+* follow good [coding guidelines](https://github.com/golang/go/wiki/CodeReviewComments).
```
Comment (veenamj-avesha) [inline CONTRIBUTING.md:48]: Change follow to Follow

File: CONTRIBUTING.md [hunk 9]
```diff
@@ -0,0 +1,55 @@
...
+* write good [commit messages](https://github.com/angular/angular/blob/main/CONTRIBUTING.md#-commit-message-format).
```
Comment (veenamj-avesha) [inline CONTRIBUTING.md:49]: Change write to Write

File: code_of_conduct.md [hunk 10]
```diff
@@ -0,0 +1,127 @@
...
+
+We pledge to act and interact in ways that contribute to an open, welcoming,
+diverse, inclusive, and healthy community.
+
+## Our Standards
+
+Examples of behavior that contributes to a positive environment for our
+community include:
+
+* Demonstrating empathy and kindness toward other people
```
Comment (veenamj-avesha) [inline code_of_conduct.md:21]: change toward to towards 

end the sentence with a period/full stop in all the bulleted list points.

File: code_of_conduct.md [hunk 11]
```diff
@@ -0,0 +1,127 @@
...
+* Being respectful of differing opinions, viewpoints, and experiences
+* Giving and gracefully accepting constructive feedback
+* Accepting responsibility and apologizing to those affected by our mistakes,
+  and learning from the experience
+* Focusing on what is best not just for us as individuals, but for the overall
+  community
+
+Examples of unacceptable behavior include:
+
```
Comment (veenamj-avesha) [inline code_of_conduct.md:30]: End the sentence with a period/full stop in all the bulleted list points.

File: code_of_conduct.md [hunk 12]
```diff
@@ -0,0 +1,127 @@
...
+an individual is officially representing the community in public spaces.
+Examples of representing our community include using an official e-mail address,
+posting via an official social media account, or acting as an appointed
+representative at an online or offline event.
+
+## Enforcement
+
+Instances of abusive, harassing, or otherwise unacceptable behavior may be
+reported to the community leaders responsible for enforcement at
+[INSERT CONTACT METHOD].
```
Comment (veenamj-avesha) [inline code_of_conduct.md:64]: Is this a placeholder to add contact details?

File: developer_guide.md [hunk 13]
```diff
@@ -0,0 +1,100 @@
...
+```
+
+2. Adjust image name variable `IMG` in the [`Makefile`](Makefile) to change the docker tag to be built.
+   Default image is set as `IMG ?= aveshasystems/worker-operator:latest`. Modify this if required.
+
+```bash
+make docker-build
+```
+
+3. Loading kubeslice-worker Image Into Your Kind Cluster ([`link`](https://kind.sigs.k8s.io/docs/user/quick-start/#loading-an-image-into-your-cluster))
```
Comment (veenamj-avesha) [inline developer_guide.md:35]: Change `Image Into Your Kind Cluster` to `image into your Kind cluster` 

and end this sentence with a period/full stop

File: developer_guide.md [hunk 14]
```diff
@@ -0,0 +1,100 @@
...
+   If needed, replace `aveshasystems/worker-operator` with your locally built image name in the previous step.
+   [See loading an image into your cluster](https://kind.sigs.k8s.io/docs/user/quick-start/#loading-an-image-into-your-cluster)
```
Comment (veenamj-avesha) [inline developer_guide.md:37]: End this sentence with a period/full stop.

File: developer_guide.md [hunk 15]
```diff
@@ -0,0 +1,100 @@
...
+```bash
+kind load docker-image aveshasystems/worker-operator --name kind
+```
+
+4. Check the loaded image in the cluster. Modify node name if required.
```
Comment (veenamj-avesha) [inline developer_guide.md:42]: Change - Modify the node name ...

File: developer_guide.md [hunk 16]
```diff
@@ -0,0 +1,100 @@
...
+
+```bash
+make test
+```
+
+### Uninstalling the kubeslice-worker
+
+Refer to the [uninstall guide](https://docs.avesha.io/opensource/uninstalling-kubeslice)
+
+1. [Detach](https://docs.avesha.io/opensource/detaching-the-applications) the application from the slice.
```
Comment (veenamj-avesha) [inline developer_guide.md:87]: detaching the applications will not be supported. So we need to change it to `offboarding namespaces` 

cc @gourishkb / @rajendra-avesha 

This applies to the controller contributor guide if detaching applications is described as a step - @narmidm

File: CONTRIBUTING.md (hunk 3 above)
Comment (gourishkb) [reply CONTRIBUTING.md:16]: @rajendra-avesha can comment on this.

File: code_of_conduct.md (hunk 12 above)
Comment (gourishkb) [reply code_of_conduct.md:64]: yes, it is. Need advise on this @pnavali @rajendra-avesha

File: developer_guide.md (hunk 16 above)
Comment (gourishkb) [reply developer_guide.md:87]: @veenamj-avesha I am using the same title as in the official docs. I think this is a prerequisite for cleaning up properly. 
Do you think it should be renamed to ```offboarding namespaces```

**Review by kmjayadeep — APPROVED** [review]

**Comment by kmjayadeep** [comment]
Let's merge this and raise follow up PRs if necessary

**Pull request merged**